[nikil] Subscribe backend

carstn opened the pull request.

Commit:
Add route/function: subscribe student to class

Commit:
Add error handling to class.subscribe

nikilselvam commented on routes/class.js:
```diff
@@ -24,4 +24,31 @@ exports.create = function(req, res){
+    Class.findOne({ _id:req.classId }, function (err, class) {
+        if (err) return console.error(err);
+        
+        class.studentIds.push(res.studentId);
```
Oh, I think this should reference `req.studentId` rather than `res.studentId'

nikilselvam commented on routes/class.js:
```diff
@@ -24,4 +24,31 @@ exports.create = function(req, res){
+    
+    // Add the Class._id to Student.classes
+    Student.findOne({ _id:req.studentId }, function (err, student) {
+        student.classes.push(res.classId);
```
Same here with `req` in place of `res`!

nikilselvam commented on routes/class.js:
```diff
@@ -24,4 +24,31 @@ exports.create = function(req, res){
+    });
+    
+    // Add the Class._id to Student.classes
+    Student.findOne({ _id:req.studentId }, function (err, student) {
```
I think this can function can placed in the callback of `class.save()` above after the error checking, since that'll give us the freedom to only add the class to the student if the student was successfully added to the class.

nikilselvam commented:
Done reviewing! I found a few places where `req` should be used in place of `res`, and I think we can include the second `findOne` function in the callback of `class.save()`. Otherwise I think this should be good to go afterwards.

Commit:
Add database testing environment

In secrets, write:
exports.testingUser = 'Carsten';

(use your own first name, first letter uppercase)

Commit:
fix test db environment testing

Commit:
Fix subscribe function

Removed an extra require.
Changed mistaken ‘res’ to ‘req’.
Changed name of ‘class’ object because it’s a reserved word.

Commit:
fix testEnv reference in db.js

Commit:
Merge branch 'master' into subscribe-backend

nikilselvam commented on routes/class.js:
```diff
@@ -2,26 +2,56 @@ var db = require('./../db.js');
+    }, function (err, classObj) {
+        if (err) return console.error(err);
+
+        classObj.studentIds.push(res.studentId);
```
We'll want to use `req.studentId` here rather than `res.studentId`!

nikilselvam commented on routes/class.js:
```diff
@@ -2,26 +2,56 @@ var db = require('./../db.js');
+            Student.findOne({
+                _id: req.studentId
+            }, function (err, student) {
+                student.classes.push(res.classId);
```
Same as above!

nikilselvam commented:
@carstn We'll need to make some `res` to `req` changes that I've specified in the code, but otherwise this should be good to merge

Commit:
fixed res -> req

Commit:
disabled test db (on digitalocean) by default

carstn commented:
merging this
